### PR TITLE
Update the error message validation for .txt tests

### PR DESCRIPTION
**Pull request by nguhoa**
Fix issue https://github.com/microsoft/Power-Fx-Dataverse/issues/219.
Update logic of error message validation for .txt tests to make the order of message sentences does not matter.

### CODE REVIEW

File: src/tests/Microsoft.PowerFx.Core.Tests/ExpressionTestCases/Mod_Float.txt [hunk 1]
```diff
@@ -32,4 +32,4 @@
 -5.8180083079168396E-201
 
 >> Mod(1E+400, 3)
-Errors: Error 4-10: Numeric value is too large.|Error 0-14: The function 'Mod' has some invalid arguments.|Error 4-10: Invalid argument type (Error). Expecting a Number value instead.
```
Comment (MikeStall) [inline src/tests/Microsoft.PowerFx.Core.Tests/ExpressionTestCases/Mod_Float.txt:35]: With the more fleixble comparsion in baserunner.cs, are these changes needed now?

Comment (nguhoa) [reply src/tests/Microsoft.PowerFx.Core.Tests/ExpressionTestCases/Mod_Float.txt:35]: not need, just for testing that even message return are different, we still can pass the check.

Comment (nguhoa) [reply src/tests/Microsoft.PowerFx.Core.Tests/ExpressionTestCases/Mod_Float.txt:35]: Maybe i just modify Mod test and no change for text_format to test this change here.

File: src/tests/Microsoft.PowerFx.Core.Tests/ExpressionTestHelpers/BaseRunner.cs [hunk 2]
```diff
@@ -208,19 +208,31 @@ private async Task<(TestResult, string)> RunAsync2(TestCase testCase)
...
                         // for tests that are run with and without NumberIsFloat set.
-                        if (actualStr.Contains(expected))
+                        if (isValid)
                         {
                             // Compiler errors result in exceptions
                             return (TestResult.Pass, null);
                         }
-                        else if (NumberIsFloat && expected.StartsWith("Errors:") &&
-                                 actualStr.Contains(Regex.Replace(expected, "(?<!Number,)(\\s|'|\\()Decimal(\\s|'|,|\\.|\\))", "$1Number$2")))
+                        else if (NumberIsFloat && actualStr.Contains(Regex.Replace(expected, "(?<!Number,)(\\s|'|\\()Decimal(\\s|'|,|\\.|\\))", "$1Number$2")))
```
Comment (CarlosFigueiraMSFT) [inline src/tests/Microsoft.PowerFx.Core.Tests/ExpressionTestHelpers/BaseRunner.cs:235]: >NumberIsFloat && actualStr.Contains(Regex.Replace(expected, "(?<!Number,)(\\s|'|\\()Decimal(\\s|'|,|\\.|\\))", "$1Number$2") [](http://example.com/codeflow?start=33&length=124)

We should move this check to inside the loop (line 218) - this way it will work with out-of-order messages as well

**Comment by LucGenetier** [comment]
✅ No public API change.

File: src/tests/Microsoft.PowerFx.Core.Tests/ExpressionTestHelpers/BaseRunner.cs (hunk 2 above)
Comment (nguhoa) [reply src/tests/Microsoft.PowerFx.Core.Tests/ExpressionTestHelpers/BaseRunner.cs:235]: updated.

**Comment by LucGenetier** [comment]
✅ No public API change.

**Review by CarlosFigueiraMSFT — APPROVED** [review]
🚢

**Comment by LucGenetier** [comment]
✅ No public API change.
